src/main/native/cpp/networktables/NetworkTableInstance.cpp

```diff
@@ -0,0 +1,62 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) FIRST 2017. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+#include "networktables/NetworkTableInstance.h"
+
+#include "llvm/SmallString.h"
+
+using namespace nt;
+
+std::shared_ptr<NetworkTable> NetworkTableInstance::GetTable(
+    StringRef key) const {
+  if (key.empty() || key[0] == NetworkTable::PATH_SEPARATOR_CHAR) {
+    return std::make_shared<NetworkTable>(m_handle, key);
+  } else {
+    llvm::SmallString<128> path;
+    path += NetworkTable::PATH_SEPARATOR_CHAR;
+    path += key;
+    return std::make_shared<NetworkTable>(m_handle, path);
+  }
+}
+
+void NetworkTableInstance::StartClient(ArrayRef<StringRef> servers,
+                                       unsigned int port) {
+  llvm::SmallVector<std::pair<StringRef, unsigned int>, 8> server_ports;
+  for (const auto& server : servers)
+    server_ports.emplace_back(std::make_pair(server, port));
+  StartClient(server_ports);
+}
+
+void NetworkTableInstance::SetServer(ArrayRef<StringRef> servers,
+                                     unsigned int port) {
+  llvm::SmallVector<std::pair<StringRef, unsigned int>, 8> server_ports;
+  for (const auto& server : servers)
+    server_ports.emplace_back(std::make_pair(server, port));
+  SetServer(server_ports);
+}
+
+NT_EntryListener NetworkTableInstance::AddEntryListener(
+    StringRef prefix, EntryListener callback, unsigned int flags) const {
+  return ::nt::AddEntryListener(
+      m_handle, prefix,
+      [=](NT_EntryListener /*uid*/, NT_Entry entry, StringRef /*name*/,
+          std::shared_ptr<Value> value, unsigned int flags_) {
+        callback(NetworkTableEntry{entry}, value, flags_);
+      },
+      flags);
+}
+
+NT_ConnectionListener NetworkTableInstance::AddConnectionListener(
+    ConnectionListener callback, bool immediate_notify) const {
+  return ::nt::AddConnectionListener(
+      m_handle,
+      [=](NT_ConnectionListener
+          /*uid*/,
```

Comment:
Yes.  It is a common idiom to comment out the parameter variable name when the parameter is unused in the function, rather than just deleting the parameter name entirely.